script/lib/filesystem.py

```diff
@@ -51,15 +51,18 @@ def download_and_extract(destination, url):
     with contextlib.closing(urllib2.urlopen(url)) as u:
       total = int(u.headers['content-length'])
       done = 0
+      last_length = 0
       while True:
         chunk = u.read(1024)
         done += len(chunk)
         if not len(chunk):
           break
-        sys.stderr.write('                                   \r')
-        percent = str(round(float(done) / float(total), 4) * 100) + '%'
+        percent = '{0:.2f}'.format(round(float(done) / float(total), 4) * 100) + '%'
         ratio = '(' + byte_to_mb(done) + '/' + byte_to_mb(total) + ')'
-        sys.stderr.write(percent + ' ' + ratio + '\r')
+        line = '-> ' + percent + ' ' + ratio
+        c_length = last_length
+        last_length = len(line)
+        sys.stderr.write(line.ljust(c_length) + '\r')
```

Comment:
`c_length` is unnecessary if you swap the write and calculate lines:

```python
line = ...
sys.stderr.write(line.ljust(last_length) + '\r')
last_length = len(line)
```

Reply:
I felt there would be a way to do that.... Mind blanked at midnight 😆